src/main/java/com/box/sdk/BoxAPIResponseException.java

```diff
@@ -40,16 +40,12 @@ public BoxAPIResponseException(String message, BoxAPIResponse responseObj) {
         try {
             responseJSON = JsonObject.readFrom(responseObj.bodyToString());
 
-            if (responseObj.bodyToString() != null && responseJSON.get("request_id") != null
-                    || this.getHeaders().containsKey("BOX-REQUEST-ID")) {
-
-                if (responseJSON.get("request_id") != null) {
-                    requestId += responseJSON.get("request_id").asString();
-                }
+            if (responseObj.bodyToString() != null && responseJSON.get("request_id") != null) {
+                requestId += responseJSON.get("request_id").asString();
+            }
 
-                if (this.getHeaders().containsKey("BOX-REQUEST-ID")) {
-                    requestId += "." + this.getHeaders().get("BOX-REQUEST-ID").get(0).toString();
-                }
+            if (responseObj.bodyToString() != null && this.getHeaders().containsKey("BOX-REQUEST-ID")) {
```

Comment:
I think you shouldn't need to check that the body is non-null here, since you're just reading out of the headers.  Is there a reason to keep this check here?